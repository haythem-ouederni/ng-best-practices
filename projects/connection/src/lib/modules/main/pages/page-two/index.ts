export * from './page-two.component';
