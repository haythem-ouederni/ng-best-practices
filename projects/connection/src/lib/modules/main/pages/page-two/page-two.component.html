<div class="alert alert-success" role="alert">
  {{'cnx.page_two.title' | translate }}
</div>

<div>
  {{'cnx.page_two.labels.username' | translate }} : <span class="badge badge-primary">{{username$ | async}}</span>
</div>

<div>
  {{'cnx.page_two.labels.password' | translate }} : <span class="badge badge-secondary">{{password$ | async}}</span>
</div>

<div>
  {{'cnx.page_two.labels.age' | translate }} : <span class="badge badge-success">{{age$ | async}}</span>
</div>

<div>
  <p>{{'cnx.page_two.labels.all_information' | translate }} :</p>
  <p><span class="badge badge-info">{{userInformation$ | async | json}}</span></p>
</div>

<div>
  <button (click)="goToPageOne()" class="btn btn-success" type="button">
    {{'cnx.page_two.buttons.go_to_page_1' | translate }}
  </button>
</div>
