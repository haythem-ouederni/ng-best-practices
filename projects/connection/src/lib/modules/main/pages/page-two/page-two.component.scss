:host {
  text-align: center;
}

div {
  padding: 15px;
}
