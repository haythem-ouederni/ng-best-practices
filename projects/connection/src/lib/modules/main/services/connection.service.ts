import {Injectable} from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class ConnectionService {
  constructor() {}

  calcultate(): string {
    return 'testing man';
  }
}
