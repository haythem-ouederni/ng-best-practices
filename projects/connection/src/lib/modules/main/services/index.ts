export * from './token';
export * from './connection.service';
