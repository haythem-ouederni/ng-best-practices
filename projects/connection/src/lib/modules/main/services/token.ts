import {InjectionToken} from '@angular/core';

export const CONNECTION_PATH = new InjectionToken<string>('CONNECTION_PATH');
