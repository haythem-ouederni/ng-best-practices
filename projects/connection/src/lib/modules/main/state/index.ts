export * from './connection.action';
export * from './connection.state';
export * from './connection.facade';
