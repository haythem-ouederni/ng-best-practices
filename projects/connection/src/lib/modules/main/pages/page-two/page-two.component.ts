import {Component, Inject, OnInit} from '@angular/core';
import {ROUTES_PATHS} from '../../connection.constant';
import {CONNECTION_PATH} from '../../services/token';

@Component({
  selector: 'cnx-page-two',
  templateUrl: './page-two.component.html',
  styleUrls: ['./page-two.component.scss'],
})
export class PageTwoComponent implements OnInit {
  linkToPageOne = `/${this.connectionPath}/${ROUTES_PATHS.pageOne}`;

  constructor(@Inject(CONNECTION_PATH) private connectionPath: string) {}

  ngOnInit() {}
}
