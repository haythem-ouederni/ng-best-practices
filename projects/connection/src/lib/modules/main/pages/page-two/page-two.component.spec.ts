import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import * as TRANSLATIONS_EN from '@i18n/connection/en.json';
import {Store} from '@ngxs/store';
import {TranslateTestingModule} from 'ngx-translate-testing';
import {CONNECTION_PATH} from '../../services/token';
import {ConnectionFacade} from '../../state';
import {PageTwoComponent} from './page-two.component';

describe('PageTwoComponent', () => {
  let component: PageTwoComponent;
  let fixture: ComponentFixture<PageTwoComponent>;

  beforeEach(async(() => {
    const storeMock = jest.fn();

    const facadeMock = jest.fn();

    TestBed.configureTestingModule({
      imports: [RouterTestingModule, TranslateTestingModule.withTranslations('fr', TRANSLATIONS_EN)],
      declarations: [PageTwoComponent],
      providers: [
        {
          provide: CONNECTION_PATH,
          useValue: '',
        },
        {
          provide: Store,
          useValue: storeMock,
        },
        {
          provide: ConnectionFacade,
          useValue: facadeMock,
        },
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PageTwoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
