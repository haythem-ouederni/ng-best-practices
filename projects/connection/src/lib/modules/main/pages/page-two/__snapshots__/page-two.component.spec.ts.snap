// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTwoComponent should match snapshot 1`] = `
<cnx-page-two
  age$="undefined"
  facade={[Function Function]}
  password$="undefined"
  userInformation$="undefined"
  username$="undefined"
>
  <div
    class="alert alert-success"
    role="alert"
  >
     User Information Recap
  
  </div><div>
     Direct username value : 
    <span
      class="badge badge-primary"
    >
      
    </span>
  </div><div>
     Direct password value : 
    <span
      class="badge badge-secondary"
    >
      
    </span>
  </div><div>
     Direct age value : 
    <span
      class="badge badge-success"
    >
      
    </span>
  </div><div>
    <p>
      All the user information from the state at once :
    </p>
    <p>
      <span
        class="badge badge-info"
      >
        null
      </span>
    </p>
  </div><div>
    <button
      class="btn btn-success"
      type="button"
    >
       Go to page one 
    </button>
  </div>
</cnx-page-two>
`;
